File: mvvm-java/data/src/main/java/com/wasisto/githubuserfinder/android/data/github/GithubService.java
package com.wasisto.githubuserfinder.android.data.github;

import com.wasisto.githubuserfinder.domain.models.SearchUsersResult;
import com.wasisto.githubuserfinder.domain.models.User;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

interface GithubService {

    @GET("search/users")
    Call<SearchUsersResult> searchUsers(@Query("q") String query);

    @GET("users/{username}")
    Call<User> getUser(@Path("username") String username);
}

File: mvvm-kotlin/data/src/main/java/com/wasisto/githubuserfinder/android/data/github/GithubService.kt
package com.wasisto.githubuserfinder.android.data.github

import com.wasisto.githubuserfinder.domain.models.SearchUsersResult
import com.wasisto.githubuserfinder.domain.models.User
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

internal interface GithubService {

    @GET("search/users")
    fun searchUsers(@Query("q") query: String): Call<SearchUsersResult>

    @GET("users/{username}")
    fun getUser(@Path("username") username: String): Call<User>
}